XivelySDKAndroid/src/main/java/com/xively/internal/rest/provision/ProvisioningApiFactory.java

```diff
@@ -0,0 +1,57 @@
+package com.xively.internal.rest.provision;
+
+
+import com.xively.XiSdkConfig;
+import com.xively.internal.Config;
+import com.xively.internal.logger.LMILog;
+import com.xively.sdk.BuildConfig;
+
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+
+public class ProvisioningApiFactory {
+    private StartAssociationWithCode associationApi;
+
+    private ProvisioningApiFactory() {
```

Comment:
Maybe it would be useful to create a template of these methods, because it is very similar in format with other factories.

Reply:
I've cleaned out a small part of it but right now I cannot give a generic solution because of the dynamic amount of API creation in each constructor especially in BP. :/